src/main_thread/decrypt/utils/check_key_statuses.ts

```diff
@@ -211,9 +204,35 @@ export default function checkKeyStatuses(
           break;
         }
 
-        default:
+        case "usable-in-future": {
+          // those key are not usable now
+          blacklistedKeyIds.push(keyId);
+          break;
+        }
+
+        case "usable": {
           whitelistedKeyIds.push(keyId);
           break;
+        }
+
+        case "status-pending": {
+          // keys are not whitelisted nor blacklisted
+          // because we don't know yet if they are usable.
+          break;
+        }
+
+        case "output-downscaled":
+        case "released": {
+          const error = new EncryptedMediaError(
+            "KEY_STATUS_CHANGE_ERROR",
+            `A "${keyStatus}" status has been encountered (${bytesToHex(keyId)})`,
+            { keyStatuses: [keyStatusObj, ...badKeyStatuses] },
+          );
+          throw error;
```

Comment:
I don't know if "output-downscaled" should be considered an error, for resiliency.

If it is output-downscaled, it's theoretically playing, just at a lower quality, isn't that semi-OK (like a warning-type of scenario?)

Reply:
Thanks I have updated the behavior to authorized to use a key that is `"output-downscaled"`.

I'm wondering if we should take in consideration that the key status is `"output-downscaled"` to select the appropriate track.
For example a content with 2 representations:
- Representation 1:  FHD (1080p) with key status `usable`.
- Representation 2: UHD (4K) with key status `output-downscaled`.

Currently the player will select the highest quality (UHD) if the client has the required bandwidth.

But which representation has a better perceived quality between the downscaled UHD and the FHD ?

Could it be possible that the downscaled UHD representation would be at a worse quality than the FHD representation ? In it's case we don't want to select the highest quality because it has been downscaled.

Reply:
`output-downscaled` doesn't seem to indicate details about the downscaling it performs, so I'm not sure if we could be able to compare it with other Representations (if there even is other Representations).

Treating "output-downscaled" as a "whitelisted" case seems ok to me in the meantime, as we've never yet even encountered that key status, but we might have to set a more complex logic in the future if we do observe some unwanted situation with it.